### DISCUSSION TITLE
Is it possible to convert GRIB files to NetCDF with nctoolkit?

### DISCUSSION THREAD

**Discussion by JairoVS** [pmlmodelling/nctoolkit#116] [Q&A]
Hello! I need to convert GRIB files into NetCDF format, would it be possible to do it using nctoolkit? I could not find a method in the "read the docs". I appreciate your help!

**Comment by robertjwilson** [comment]
Right now there is no built in method for this. However, you can use CDO to do it, by calling it from nctoolkit as follows:

    ds.cdo_command("-f nc copy")

In technical terms, nctoolkit could provide complete support for grib files, as CDO can handle them fine. The tricky issue is that some methods (such as plotting) require netCDF files, so it's hard to do things efficiently as you may or may not need to convert to netcdf in a processing chain. I'll leave this open so I can figure out a smart way to implement this.

**Reply by JairoVS** [reply]
I appreciate your answer Robert. I will try working directly with the GRIB files then and see how it goes. The thing is I will work with a large dataset of monthly meteorological variables, so maybe the best would be to convert them from the beginning so I can apply all the nctoolkit methods smoothly, including plotting.

**Reply by robertjwilson** [reply]
It's probably safest to convert to netCDF first. Interally, (for a few methods) nctoolkit will occassionally call Python netCDF libraries so you might throw errors if you try to work directly with the grib files.

**Reply by JairoVS** [reply]
I will follow your advice and convert the files to netCDF.
